lib/plugins/inventory.js

```diff
@@ -30,11 +30,27 @@ function inject (bot, { version, hideErrors }) {
   bot.inventory = windows.createWindow(0, 'minecraft:inventory', 'Inventory')
   bot.currentWindow = null
   bot.heldItem = null
+  bot.usingHeldItem = false
 
   bot._client.on('entity_status', (packet) => {
     if (packet.entityId === bot.entity.id && packet.entityStatus === 9 && !eatingTask.done) {
       eatingTask.finish()
     }
+    bot.usingHeldItem = false
+  })
+
+  bot.on('heldItemChanged', () => {
+    if (!eatingTask.done) {
+      eatingTask.finish()
+    }
+    bot.usingHeldItem = false
+  })
+
+  bot._client.on('set_cooldown', () => {
+    if (!eatingTask.done) {
+      eatingTask.finish()
+    }
+    bot.usingHeldItem = false
   })
```

Comment:
I would check that the item id the set_cooldown packet is referring to is the same as the item id of the item the bot is holding as it appears to be used only with ender pearls and chorus https://wiki.vg/Protocol#Set_Cooldown

Something like this under line 49 would do:
```js
if (bot.heldItem && bot.heldItem.type !== packet.itemID) return
```

Reply:
Alright I changed that. Also the packet is also used when someone disables your shield with an axe.